src/utils/stable-sort.js

```diff
@@ -0,0 +1,24 @@
+/*
+ * Consitant and stable sort function across JavsaScript platforms
+ *
+ * Inconsistant sorts can cause SSR problems between client and server
+ * such as in <b-table> if sortBy is applied to the data on server side render.
+ * Chrome and V8 native sorts are inconsistant/unstable
+ *
+ * This function uses native sort with fallback to index compare when the a and b
+ * compare returns 0
+ *
+ * Algorithm bsaed on:
+ * https://stackoverflow.com/questions/1427608/fast-stable-sorting-algorithm-implementation-in-javascript/45422645#45422645
+ *
+ * @param {array} array to sort
+ * @param {function} sortcompare function
+ * @return {array}
+ */
+
+export default function stableSort (array, compareFn) {
+  return array
+    .map((a, index) => [ index, a ])
+    .sort(function (a, b) { return this(a[1], b[1]) || a[0] - b[0] }.bind(compareFn))
```

Comment:
This works the same way:
`.sort( (a, b) => compareFn(a[1], b[1]) || (a[0] - b[0]) )`

Reply:
Binding the compareFn as the `this` context speeds up performance about 5%

Reply:
did you test it? o_O

Reply:
with arrow funcs stableSort execution times between 0,21 - 1 ms.
with bind(): 0,32 - 1.4 ms

Reply:
In chrome

Reply:
and overall sort-click behavior average execution time less with arrow funcs

Reply:
`.sort((a, b) => compareFn(a[1], b[1]) || (a[0] - b[0]))`
![image](https://user-images.githubusercontent.com/20986566/33193815-2e822250-d0dc-11e7-8208-17d98bd00e12.png)

`.sort(function (a, b) { return this(a[1], b[1]) || a[0] - b[0] }.bind(compareFn))`
![image](https://user-images.githubusercontent.com/20986566/33193865-83883ed8-d0dc-11e7-82ed-d09057ef4115.png)

Reply:
`.sort(function (a, b) { return this(a[1], b[1]) || a[0] - b[0] }.bind(compareFn))`
![bind](https://user-images.githubusercontent.com/20986566/33194127-b95d9a42-d0de-11e7-9339-8b11e60f1c06.PNG)

`.sort((a, b) => compareFn(a[1], b[1]) || (a[0] - b[0]))`
![arrow](https://user-images.githubusercontent.com/20986566/33194138-cd087efe-d0de-11e7-9157-3ca5c51c7de7.PNG)

Reply:
Based on the original source: https://stackoverflow.com/questions/1427608/fast-stable-sorting-algorithm-implementation-in-javascript/45422645#45422645
>Using .bind(compareFunction) on the wrapped anonymous function within stableSort() boosted that relative performance from about 38% by avoiding an unneeded scoped reference to compareFunction on each call by assigning it to the context instead.

Reply:
It compares with native `Array.sort()`, not the arrow functions implementation. 
And besides, after babel arrow funcs wil get transpiled to bind.... 
It's all about readability

Reply:
But the babel transpile will not bind the `compareFn` function as `this`, babel (transpiling the arrow function) will set `this` to the `this` context of the location it was defined, which is the outer `stableSort` function context.

This code explicitly sets `this` as the `compareFn`, so saving the function call setup on each iteration of the inner loop.

Since this function will be called every time the table needs updating (when there is a sort change), speed makes the difference with large datasets.

Reply:
https://google.github.io/styleguide/jsguide.html#features-functions-arrow-functions

Reply:
> Note that `this` in a method shorthand or function refers to the object literal itself whereas **this in an arrow function refers to the scope _outside_ the object literal.**

In this case, we do not want `this` to refer to the `stableSort` context (the outer function), we want `this` to refer to the `compareFn` (so calling `this(a, b)` actually calls `compareFn`, without the overhead of calling `compareFn` each iteration fo the `.sort(..)` routine)

Reply:
I have added  comment above the code to explain the reasoning.

Reply:
Note that a stables sort is *much* less efficient than using QuickSort (which is what V8 and Chrome use if the array size is > 10), so we are taking a larger performance hit by implementing stable sort in order to avoid SSR errors.  So being able to squeeze some speed improvements is needed.